src/components/common/GuestProtectedRoute.jsx

```diff
@@ -1,14 +1,14 @@
 import { Navigate } from 'react-router';
-import { useSelector } from 'react-redux';
+import useStorageValue from '../../hooks/useStorageValues';
 
 const GuestProtectedRoute = ({ children }) => {
```

Comment:
why do we need this one at all

Reply:
Unlike the values ​​we get from redux, the component is not subscribed to changes in `localStorage` by default.
I wrote a hook to solve this problem.